### PR TITLE
cmake: small build related fixes

### PR DESCRIPTION
**Pull request by furiel**
I ran into a small trouble while compiling syslog-ng using cmake on osx. With these patches applied, I could start syslog-ng with an example-msg-generator and stdout destination.

### CODE REVIEW

**Comment by kira-syslogng** [comment]
Build SUCCESS

**Comment by bazsi** [comment]
The amqp build fails in devshell when compiling with cake, because it doesn't test for a deprecated function call.

The same works in automake.

**Review by bazsi — APPROVED** [review]
This looks good to me, except for the broken build in devshell

**Comment by furiel** [comment]
@bazsi thank for the analysis. I will look into the devshell failure.

**Comment by kira-syslogng** [comment]
Build SUCCESS

**Review by MrAnno — APPROVED** [review]
Yay!

File: CMakeLists.txt [hunk 1]
```diff
@@ -79,6 +79,8 @@ include(CheckCSourceCompiles)
 
 add_definitions(-D_GNU_SOURCE=1)
 add_definitions(-D_LARGEFILE64_SOURCE=1)
+add_definitions(-D__APPLE_USE_RFC_3542)
```
Comment (MrAnno) [inline CMakeLists.txt:82]: We might add a conditional around this macro next time, so it won't appear on irrelevant platforms.

**Review by gaborznagy — APPROVED** [review]

**Pull request merged**